org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -56,55 +60,84 @@ public boolean performFinish() {
 		}
 
 		if (!template.isEmpty()) {
-			String createCommand = "dotnet new " + template + " -o " + location.toString();
-
-			Runtime runtime = Runtime.getRuntime();
 			try {
-				Process process = runtime.exec(createCommand);
-				if (process.waitFor() != 0) {
-					MessageDialog.openError(getShell(), "Cannot create dotnet template",
-							"The 'dotnet new' command failed.");
-					return false;
-				}
-			} catch (IOException | InterruptedException e) {
+				getContainer().run(true, true, monitor -> {
+					monitor.beginTask("Creating .NET project",0);
+					String createCommand = "dotnet new " + template + " -o " + location.toString();
+					Runtime runtime = Runtime.getRuntime();
+
+					try {
+						Process process = runtime.exec(createCommand);
+						boolean isProcessDone = false;
+						while (!isProcessDone) {
+							if (monitor.isCanceled()) {
+								process.destroyForcibly();
+							}
+							isProcessDone = process.waitFor(100, TimeUnit.MILLISECONDS);
+						}
+						monitor.done();
+						if (process.exitValue() != 0) {
+							MessageDialog.openError(getShell(), "Cannot create dotnet template",
+									"The 'dotnet new' command exited with :" + process.exitValue());
+						} else {
+							createProject(projectName, location);
+						}
+					} catch (IOException | InterruptedException e) {
+						monitor.done();
+						MessageDialog.openError(getShell(), "Cannot create dotnet template",
+								"The 'dotnet new' command failed.");
+					}
+				});
+			} catch (InvocationTargetException | InterruptedException e) {
 				MessageDialog.openError(getShell(), "Cannot create dotnet template",
-						"The 'dotnet new' command failed.");
-				return false;
+						"The 'dotnet new' command failed");
 			}
+		} else {
+			createProject(projectName, location);
 		}
+		return true;
+	}
+
+	private void createProject(String name, File directory) {
```

Comment:
This method is actually a long operation (see many of the methods it calls do take a progressMonitor as parameter. This is usually a sign that it can block UI, so it should take a progress monitor as parameter and pass it (or a SubMonitor) to the method it calls that accept it.